modules/ocf_puppet/manifests/puppetmaster.pp

```diff
@@ -1,52 +1,75 @@
 class ocf_puppet::puppetmaster {
   package {
-    ['puppetmaster-passenger', 'puppet-lint']:;
+    ['puppetserver', 'puppet-lint']:;
   }
 
-  class { '::apache':
-    default_vhost => false;
+  service { 'puppetserver':
+    enable  => true,
+    require => Package['puppetserver'],
   }
 
-  apache::vhost { 'puppetmaster':
-    docroot    => '/usr/share/puppet/rack/puppetmasterd/public/',
-    port       => 8140,
-
-    ssl               => true,
-    ssl_key           => '/var/lib/puppet/ssl/private_keys/puppet.pem',
-    ssl_cert          => '/var/lib/puppet/ssl/certs/puppet.pem',
-    ssl_chain         => '/var/lib/puppet/ssl/certs/ca.pem',
-    ssl_ca            => '/var/lib/puppet/ssl/certs/ca.pem',
-    ssl_crl           => '/var/lib/puppet/ssl/ca/ca_crl.pem',
-    ssl_verify_client => 'optional',
-    ssl_verify_depth  => 1,
-    ssl_options       => ['+StdEnvVars', '+ExportCertData'],
-
-    rack_base_uris => ['/'];
+  # Set correct memory limits on puppetserver so that it doesn't run out
+  augeas { '/etc/default/puppetserver':
+    context => '/files/etc/default/puppetserver',
+    changes => [
+      "set JAVA_ARGS '\"-Xms1g -Xmx1g -XX:MaxPermSize=512m\"'",
+    ],
```

Comment:
Yeah, not a fan of the memory usage and startup time of puppet server, but it does serve requests faster, so I guess that's a reasonable tradeoff? /shrug